FIAT/gauss_legendre.py

```diff
@@ -31,3 +36,16 @@ def __init__(self, ref_el, degree):
         dual = GaussLegendreDualSet(ref_el, degree)
         formdegree = ref_el.get_spatial_dimension()  # n-form
         super(GaussLegendre, self).__init__(poly_set, dual, degree, formdegree)
+
+    def tabulate(self, order, points, entity=None):
+        # This overrides the default with a more numerically stable algorithm
+
+        if entity is None:
+            entity = (self.ref_el.get_dimension(), 0)
+
+        entity_dim, entity_id = entity
+        transform = self.ref_el.get_entity_transform(entity_dim, entity_id)
+
+        xsrc = numpy.array([list(node.get_point_dict())[0][0] for node in self.dual.nodes])
```

Comment:
Although it's unlikely to change, we should probably write this so that it's safe in the case that FIAT ever moves to moments for the nodes.

```python
xsrc = numpy.empty(len(self.dual.nodes), dtype=double)
for i, node in enumerate(self.dual.nodes):
    # assert that we have a singleton point
    (pt,), = node.get_point_dict().keys()
    xsrc[i] = pt
```